myst_nbs/case_studies/wrapping_jax_function.myst.md

```diff
@@ -0,0 +1,765 @@
+    logp_initial_state_true,
+    logp_transition_true,
+)
+jax_fn = aesara.function(inputs=[], outputs=out, mode="JAX")
+jax_fn()
+```
+
+We can also compile a JAX function that computes the log probability of each variable in our PyMC model, similar to {meth}`~pymc.Model.point_logps`. We will use the helper method {meth}`~pymc.Model.compile_fn`.
+
+```{code-cell} ipython3
+model_logp_jax_fn = model.compile_fn(model.logpt(sum=False), mode="JAX")
+model_logp_jax_fn(initial_point)
+```
+
+Note that we could have added an equally simple function to convert our `HMMLogpGradOp`, in case we wanted to convert Aesara gradient graphs to JAX. In our case, we don't need to do this because we will rely on JAX `grad` function (or more precisely, NumPyro will rely on it) to obtain these again from our compiled JAX function.
+
+We include a {ref}`short discussion <aesara_vs_jax>` at the end of this document, to help you better understand the trade-offs between working with Aesara graphs vs JAX functions, and when you might want to use one or the other.
+
++++
+
+### Sampling with NumPyro
+
++++
+
+Now that we know our model logp can be entirely compiled to JAX, we can use the handy {mod}`pymc.sampling_jax` module to sample our model using the pure JAX sampler implemented in NumPyro.
+
+```{code-cell} ipython3
+with model:
+    idata_numpyro = pm.sampling_jax.sample_numpyro_nuts(chains=2, progress_bar=False)
+```
+
+```{code-cell} ipython3
+az.plot_trace(idata_numpyro);
+```
+
+```{code-cell} ipython3
+az.plot_posterior(idata_numpyro, ref_val=true_values, grid=(3, 5));
+```
+
+As expected, sampling results look pretty similar! 
+
+Depending on the model and computer architecture you are using, a pure JAX sampler can provide considerable speedups.
+
++++
+
+(aesara_vs_jax)=
+## Some brief notes on using Aesara vs JAX
+
++++
+
+### When should you use JAX?
+
++++
+
+As we have seen, it is pretty straightforward to interface between Aesara graphs and JAX functions. 
+
+This can be very handy when you want to combine previously implemented JAX function with PyMC models. We used a marginalized HMM log-likelihood in this example, but the same strategy could be used to do Bayesian inference with Deep Neural Networks or Differential Equations, or pretty much any other functions implemented in JAX that can be used in the context of a Bayesian model.
+
+It can also be worth it, if you need to make use of JAX's unique features like vectorization, support for tree structures, or its fine-grained parallelization, and GPU and TPU capabilities.
+
++++
+
+### When should you not use JAX?
+
++++
+
+Like JAX, Aesara has the goal of mimicking the NumPy and Scipy APIs, so that writing code in Aesara should feel very similar to how code is written in those libraries.
+
+There are, however, some of advantages to working with Aesara:
+
+1. Aesara graphs are considerably easier to [inspect and debug](https://aesara.readthedocs.io/en/latest/tutorial/debug_faq.html) than JAX functions
```

Comment:
The link should be `` {ref}`inspect and debug <aesara:debug_faq>` ``. This one has a [target defined](https://raw.githubusercontent.com/aesara-devs/aesara/main/doc/tutorial/debug_faq.rst). 

Similar comments for the two below, I can help if needed.

Reply:
How do I find out which pages have "targets defined" and what those should be?

Reply:
In rST they are the `.. _target_id:` in MyST they are the `(target_id)=`. They generally come right before titles and point to that title but could also point to specific items in a bullet point or numbered list. 

Note that the target is `target_id` in both, the preceding underscore in rST is not part of the target but of the syntax.